client/notifier/channel-notifier.go

```diff
@@ -0,0 +1,90 @@
+package notifier
+
+import (
+	"github.com/statechannels/go-nitro/client/engine/store"
+	"github.com/statechannels/go-nitro/client/query"
+	"github.com/statechannels/go-nitro/internal/safesync"
+	"github.com/statechannels/go-nitro/payments"
+	"github.com/statechannels/go-nitro/types"
+)
+
+const ALL_NOTIFICATIONS = "all"
+
+// ChannelNotifier is used to notify multiple listeners of a channel update.
+type ChannelNotifier struct {
```

Comment:
I don't think we easily can, because then it becomes difficult to reference a `ChannelNotifier` at all. For example the [client wouldn't be allowed to have a pointer](https://github.com/statechannels/go-nitro/blob/0973ca819b7203f2202307a37d4c52233866c373/client/client.go#L31) to a `ChannelNotifier`

We could probably get around this by exposing some `ChannelNotifier` interface but I'm not sure it's worth the effort?